src/status.py

```diff
@@ -17,26 +25,6 @@ def get_priority_status(statuses: typing.Iterable[ops.StatusBase]) -> ops.Status
     Returns:
         The final status to display.
     """
-
-    def get_status_priority(status: ops.StatusBase) -> int:
-        """Get status priority in numerical value.
-
-        Args:
-            status: The status to convert to priority value.
-
-        Returns:
-            The status priority value integer.
-        """
-        priority = {
-            ops.ErrorStatus.name: 0,
-            ops.BlockedStatus.name: 2,
-            ops.MaintenanceStatus.name: 4,
-            ops.WaitingStatus.name: 6,
-            ops.ActiveStatus.name: 8,
-        }.get(status.name)
-        priority = typing.cast(int, priority)
-        if status.message:
-            return priority - 1
-        return priority
-
-    return sorted(statuses, key=get_status_priority)[0]
+    return sorted(statuses, key=lambda item: (PRIORITY_MAP[item.name] - int(bool(item.message))))[
```

Comment:
Close, let's change to lambda returning `(PRIORITY_MAP[item.name], int(bool(item.message)))`, check whether this will get the correct order, perhaps the second element needs a minus